internal/subscription/core.go

```diff
@@ -497,3 +501,60 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 	logger.Ctx(ctx).Infow("migration: request completed.", "subscriptionsUpdated", updatedSubCount)
 	return nil
 }
+
+func (c *Core) deleteSubscriptionTopics(ctx context.Context, m *Model) error {
+	if m == nil || m.GetTopic() == "" {
+		return errors.New("Nil model")
+	}
+
+	projectID := m.ExtractedTopicProjectID
+
+	//  delete internal topic
+	internalTopic := m.GetSubscriptionTopic()
+	err := c.deleteSubscriptionTopic(ctx, internalTopic, projectID, topic.GetTopicNameOnly(internalTopic))
+	if err != nil {
+		return err
+	}
+
+	// delete retry topic
+	retryTopic := m.GetRetryTopic()
+	err = c.deleteSubscriptionTopic(ctx, retryTopic, projectID, topic.GetTopicNameOnly(retryTopic))
+	if err != nil {
+		return err
+	}
+
+	// delete delay topics
+	for _, delayTopic := range m.GetDelayTopics() {
+		delayModel := &topic.Model{
+			Name:               delayTopic,
+			ExtractedProjectID: projectID,
+			ExtractedTopicName: topic.GetTopicNameOnly(delayTopic),
+		}
+		err = c.topicCore.DeleteSubscriptionTopic(ctx, delayModel)
```

Comment:
why don't we use c.deleteSubscriptionTopic here..?

Reply:
changes in the method